### PR TITLE
Legacy MAP paragraph text containing the <link> tag is not rendered correctly

### LINKED ISSUE #5447: Link elements are not being rendered correctly for TC2 MAPs
**Describe the bug**
`Link` elements are displayed as raw text rather than rendered as a hyperlink

**To Reproduce**
Steps to reproduce the behaviour:

1. Navigate to http://localhost:7080/mundo/video_fotos/2016/05/160502_video_apnea_record_rubridge_nezolandes_amv
2. Notice that the link is not rendered correctly:

`"Fue bastante difícil, especialmente el ascenso. Algo se me enredo después de tocar el fondo donde debía llegar metros. Fue terrible", posteó en la <link type="page"><caption> página de Facebook de Vertical Club</caption><url href="https://www.facebook.com/verticalblue/?fref=ts" platform="highweb"/></link>.`

**Expected behaviour**
For reference view https://www.bbc.com/mundo/video_fotos/2016/05/160502_video_apnea_record_rubridge_nezolandes_amv

**Screenshots**
image

**Desktop (please complete the following information):**
All devices and browsers

**Smartphone (please complete the following information):**
All devices and browsers

**Testing notes**

- [ ] This bug fix is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by karinathomasbbc**
Resolves #5447 

**Overall change:** 
A paragraph from the legacy MAP page can contain link data - here is a sample paragraph returned by AReS:

``` 
{
"text": "&quot;Fue bastante difícil, especialmente el ascenso. Algo se me enredo después de tocar el fondo donde debía llegar metros. Fue terrible&quot;, posteó en la <link type=\"page\"><caption>\n\npágina de Facebook de Vertical Club</caption><url href=\"https://www.facebook.com/verticalblue/?fref=ts\" platform=\"highweb\"/></link>.",
"markupType": "plain_text",
"type": "paragraph"
},
```

**Code changes:**
- If paragraph text contains `</link>`, then set the markup type to `candy_xml` so that it can be processed by the rich-text-transformer
- Updating tests accordingly

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.js [hunk 1]
```diff
@@ -31,6 +31,8 @@ export const processBlock = _block => {
     if (block.role === 'introduction') {
       block.text = boldWrap(block.text);
       block.markupType = 'candy_xml';
+    } else if (block.text.includes('</link>')) {
```
Comment (ryanmccombe) [inline src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.js]: Asked on slack but should post it here too - this PR only updates paragraphs - does this cover everything?  Seen the bug in bulleted lists too

Comment (RichardPK) [reply src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.js]: Looks like the `list` preprocessor imports `convertParagraph` and runs each block item through it.  `convertParagraph` runs the block through `processBlock` (this helper). The `subheadline` preprocessor also does this. Are there any other Optimo blocks we should be doing this for?

Also - is importing a paragraph helper into subheadline and list preprocessors the right way to do this? It was done a while ago, nothing to do with this PR but maybe something to discuss.

Fixes it in the bulleted list in the example - http://localhost:7080/mundo/video_fotos/2016/05/160502_video_apnea_record_rubridge_nezolandes_amv

File: src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.js [hunk 2]
```diff
@@ -27,6 +27,10 @@ export const processBlock = _block => {
   const block = _block;
 
   if (path(['text'], block)) {
+    // catch plain_text ARES response for legacy MAPs with <link> elements
+    if (block.text.includes('</link>')) {
+      block.markupType = 'candy_xml';
```
Comment (rebeccamcginn) [inline src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.js:32]: Please can you test this with a link that contains chevrons or ampersands?

Comment (RichardPK) [reply src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.js:32]: Quotations are encoded in the desired format - `&quot;` so we're optimistic that the other characters will be too. Would need to test real TC2 assets with ampersands/chevrons to see what comes back in the response but I think we're confident enough for now. Examples of the `&quot;` encoding for legacy assets below. 

`/simorgh/data/afrique/legacyAssets/nos_emissions/2014/11/141111_porno_vengeur_ouganda.json`

`/simorgh/data/persian/legacyAssets/arts/2011/08/110802_doublage_sherlock_promo.json`

`/simorgh/data/portuguese/legacyAssets/noticias/2016/03/160307_andes_vivos_tg.json`

File: src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.js [hunk 3]
```diff
@@ -27,6 +27,10 @@ export const processBlock = _block => {
   const block = _block;
```
Comment (jroebu14) [inline src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.js]: should be `const block = { ..._block }` because the way it stands code further down will mutate the `_block` that is passed in

**Review by jroebu14 — APPROVED** [review]

File: src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.test.js [hunk 4]
```diff
@@ -68,4 +68,22 @@ describe('processBlock', () => {
 
     expect(processBlock(input)).toEqual(expected);
   });
+
+  it('should convert plain_text markup to candy_xml when the <link> tag is present', () => {
+    const input = {
+      text:
+        'paragraph including link <link><caption>this is a caption</caption><url href="www.example/com" platform="highweb"/></link>',
```
Comment (karinathomasbbc) [inline src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.test.js]: Could we add a second test for the link which is in most legacy MAPs:

`<link type="page"><caption>this is a caption</caption><url href="www.example.com" platform="highweb"/></link>`

Comment (karinathomasbbc) [inline src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.test.js]: Can we fix the href - should be `www.example.com`

Comment (RichardPK) [reply src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.test.js]: Done!

Comment (RichardPK) [reply src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.test.js]: Done!

File: src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.js (hunk 3 above)
Comment (RichardPK) [reply src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.js]: Cheers Jroe, done

File: cypress/integration/pages/mediaAssetPage/tests.js [hunk 5]
```diff
@@ -51,6 +55,31 @@ export const testsThatFollowSmokeTestConfig = ({ service, pageType }) => {
       );
     });
 
+    it('if a legacy MAP it should render a link using an a tag and href, rather than a plain text <link>', () => {
+      const requestPath = config[service].pageTypes[pageType].path;
```
Comment (ryanmccombe) [inline cypress/integration/pages/mediaAssetPage/tests.js]: `path` can be an array - there's an environment variable that is a more stable way to determine the url :)

Comment (karinathomasbbc) [reply cypress/integration/pages/mediaAssetPage/tests.js]: `Cypress.env('currentPath')`?

Comment (ryanmccombe) [reply cypress/integration/pages/mediaAssetPage/tests.js]: aye that's it - sorry, had to go to a meeting to eat some chocolate oranges before I could find it 👍

**Review by ryanmccombe — APPROVED** [review]
👍

File: cypress/integration/pages/mediaAssetPage/tests.js (hunk 5 above)
Comment (RichardPK) [reply cypress/integration/pages/mediaAssetPage/tests.js]: Cheers, updated

File: cypress/integration/pages/mediaAssetPage/tests.js [hunk 6]
```diff
@@ -51,6 +55,31 @@ export const testsThatFollowSmokeTestConfig = ({ service, pageType }) => {
       );
     });
 
+    it('if a legacy MAP it should render a link using an a tag and href, rather than a plain text <link>', () => {
```
Comment (karinathomasbbc) [inline cypress/integration/pages/mediaAssetPage/tests.js]: Do you want to reword this to `legacy MAP should render a link using an <a> tag with href rather than a plain text <link>`

Comment (RichardPK) [reply cypress/integration/pages/mediaAssetPage/tests.js]: 👍
